src/nyxstone.cpp

```diff
@@ -418,10 +418,12 @@ tl::expected<void, std::string> Nyxstone::disassemble_impl(const std::vector<uin
     const llvm::ArrayRef<u8> data(bytes.data(), bytes.size());
     uint64_t pos = 0;
     uint64_t insn_count = 0;
-    while (true) {
+    uint64_t insn_size = 0;
+    std::string insn_str;
+    
+    for (; pos < data.size() && (count == 0 || insn_count < count); pos += insn_size) {
```

Comment:
I like the idea of improving the loop readability, but I think this currently doesn't help too much. I would prefer:
```
    bool disassemble_all = count == 0;
    // We exit either if we reached the end of the provided bytes, or if we have disassembled as many instructions
    // as the user has requested
    while(pos < data.size() && (disassemble_all || insn_count < count)) {
```
and iterating `insn_count` and `pos` at the end of the loop.

Reply:
Added suggested loop this is much cleaner